ReplayKitExample/ReplayKitExample/ReplayKitVideoSource.swift

```diff
@@ -145,6 +253,59 @@ class ReplayKitVideoSource: NSObject, VideoSource {
                      buffer: sourcePixelBuffer,
                      orientation: videoOrientation,
                      forceReschedule: false)
+
+        // Hold on to the previous sample buffer to prevent tearing.
+        lastSampleBuffer2 = lastSampleBuffer
+        lastSampleBuffer = sampleBuffer
+    }
+
+    /*
+     * The IVTC algorithm must know when a given frame is a duplicate of a previous frame. This implementation simply
+     * compares the chroma channels of each image to determine equality. Occasional false positives are worth the
+     * performance benefit of skipping the luma (Y) plane, which is twice the size of the chroma (UV) plane.
+     */
+    static func compareSamples(first: CMSampleBuffer, second: CMSampleBuffer) -> Bool {
+        guard let firstPixelBuffer = CMSampleBufferGetImageBuffer(first) else {
+            return false
+        }
+        guard let secondPixelBuffer = CMSampleBufferGetImageBuffer(second) else {
+            return false
+        }
+
+        // Assumption: Only NV12 is supported.
+        guard CVPixelBufferGetWidth(firstPixelBuffer) == CVPixelBufferGetWidth(secondPixelBuffer) else {
+            return false
+        }
+        guard CVPixelBufferGetHeight(firstPixelBuffer) == CVPixelBufferGetHeight(secondPixelBuffer) else {
+            return false
+        }
+
+        CVPixelBufferLockBaseAddress(firstPixelBuffer, .readOnly)
+        CVPixelBufferLockBaseAddress(secondPixelBuffer, .readOnly)
+        defer {
+            CVPixelBufferUnlockBaseAddress(firstPixelBuffer, .readOnly)
+            CVPixelBufferUnlockBaseAddress(secondPixelBuffer, .readOnly)
+        }
+
+        // Only the chroma plane is compared.
+        let planeIndex = 1
+        guard let baseAddress1 = CVPixelBufferGetBaseAddressOfPlane(firstPixelBuffer, planeIndex) else {
+            return false
+        }
+        guard let baseAddress2 = CVPixelBufferGetBaseAddressOfPlane(secondPixelBuffer, planeIndex) else {
+            return false
+        }
+        let width = CVPixelBufferGetWidthOfPlane(firstPixelBuffer, planeIndex)
+        let height = CVPixelBufferGetHeightOfPlane(firstPixelBuffer, planeIndex)
+
+        for row in 0...height {
+            let rowOffset = row * CVPixelBufferGetBytesPerRowOfPlane(firstPixelBuffer, planeIndex)
+            if memcmp(baseAddress1.advanced(by: rowOffset), baseAddress2.advanced(by: rowOffset), width) != 0 {
```

Comment:
Rather `memcmp` is there a better technique compare chroma plane?

Reply:
What I want is to be able to subtract two planes and make sure the sum of the magnitude of the resulting pixels is zero.

I looked into using some of the libraries under the accelerate umbrella, for example vDSP can do a very good job of subtracting vectors but only works with floating point data. Then there is the [simd](https://developer.apple.com/documentation/accelerate/simd) library, which could work. For example, we could iterate through each row using [simd_int4](https://developer.apple.com/documentation/simd/simd_int4).

In the end, I thought that memcmp would be sufficient since I want to compare entire rows at a time.